core/src/main/java/org/primefaces/extensions/component/clockpicker/ClockPickerRenderer.java

```diff
@@ -0,0 +1,200 @@
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package org.primefaces.extensions.component.clockpicker;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.el.ValueExpression;
+import javax.faces.application.FacesMessage;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.context.ResponseWriter;
+import javax.faces.convert.Converter;
+import javax.faces.convert.ConverterException;
+
+import org.primefaces.renderkit.CoreRenderer;
+import org.primefaces.util.LangUtils;
+import org.primefaces.util.MessageFactory;
+import org.primefaces.util.WidgetBuilder;
+
+public class ClockPickerRenderer extends CoreRenderer {
+
+    @Override
+    public void decode(FacesContext context, UIComponent component) {
+
+        final ClockPicker clockPicker = (ClockPicker) component;
+
+        if (clockPicker.isDisabled() || clockPicker.isReadonly()) {
+            return;
+        }
+        final String param = clockPicker.getClientId(context) + "_input";
+        final String submittedValue = context.getExternalContext().getRequestParameterMap().get(param);
+
+        if (null != submittedValue) {
+            clockPicker.setSubmittedValue(submittedValue);
+        }
+        decodeBehaviors(context, component);
+    }
+
+    @Override
+    public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
+        final ClockPicker clockPicker = (ClockPicker) component;
+
+        final String value = getValueAsString(context, clockPicker);
+        encodeMarkup(context, clockPicker, value);
+        encodeScript(context, clockPicker, value);
+    }
+
+    private void encodeMarkup(FacesContext context, ClockPicker clockPicker, final String value) throws IOException {
+        ResponseWriter writer = context.getResponseWriter();
+        String clientId = clockPicker.getClientId();
+        String inputId = clientId + "_input";
+
+        writer.startElement("div", clockPicker);
+        writer.writeAttribute("class", ClockPicker.CONTAINER_CLASS, null);
+
+        writer.startElement("input", null);
+        writer.writeAttribute("id", inputId, null);
+        writer.writeAttribute("name", inputId, null);
+        writer.writeAttribute("type", "text", null);
+        writer.writeAttribute("class", "ui-inputfield ui-widget ui-state-default ui-corner-all", null);
+        writer.writeAttribute("size", 5, null);
+        writer.writeAttribute("maxlength", 5, null);
```

Comment:
I am assuming size and maxlength must be 5 because that is the most there can be?  What about AM/PM?

Reply:
Yes, there's no option to input AM/PM in the clock picker component from the usage point of view: https://weareoutman.github.io/clockpicker/. It uses 24-hour format always